joblib/parallel.py

```diff
@@ -180,7 +190,7 @@ def delayed(function, check_pickle=True):
     # Try to pickle the input function, to catch the problems early when
     # using with multiprocessing:
     if check_pickle:
-        pickle.dumps(function)
+        dumps(function)
```

Comment:
cloudpickle should be able to pickle any reasonable function. Maybe we should deprecate the `check_pickle` flag and never do this check anymore.